h2o-persist-gcs/src/main/java/water/persist/PersistGcs.java

```diff
@@ -157,11 +158,18 @@ public void importFiles(String path,
       parseBucket(bk[0], files, keys, fails);
     } else {
       try {
-        Blob blob = storageProvider.getStorage().get(bk[0], bk[1]);
-        final GcsBlob gcsBlob = GcsBlob.of(blob.getBlobId());
-        final Key k = GcsFileVec.make(path, blob.getSize());
-        keys.add(k.toString());
-        files.add(path);
+        Iterable<Blob> values = storageProvider.getStorage().list(bk[0], Storage.BlobListOption.prefix(bk[1])).getValues();
+        values.forEach(blob -> {
+          final String blobPath = new StringBuilder("gs://")
```

Comment:
if you would do just plain string concat here javac will translate it to this exact code

Reply:
Ok, cool. didn't know javac will output the same bytecode. Heard of it.

Done. Thank you.